clue.js

```diff
@@ -0,0 +1,181 @@
+// STEP 1 - CREATE OBJECTS FOR ALL THE SUSPECTS, SOMETHING LIKE THIS:
+
+const mrGreen = {
+  firstName: 'Jacob',
+  lastName: 'Green',
+  color: 'green',
+  description: 'He has a lot of connections',
+  age: 45,
+  image: 'assets/green.png',
+  occupation: 'Entrepreneur'
+}
+const mrMustard = {
+  firstName: 'Colonel',
+  lastName: 'Mustard',
+  color: 'yellow',
+  description: 'He is good at weapons',
+  age: 55,
+  image: 'assets/mustard.png',
+  occupation: 'Hunter'
+}
+const mrsPeacock = {
+  firstName: 'Lena',
+  lastName: 'Peacock',
+  color: 'blue',
+  description: 'She has a lot of money',
+  age: 50,
+  image: 'assets/peacock.png',
+  occupation: 'Rich Mama'
+}
+const professorPlum = {
+  firstName: 'Lennart',
+  lastName: 'Plum',
+  color: 'purple',
+  description: 'He is a genius',
+  age: 35,
+  image: 'assets/plum.png',
+  occupation: 'Genius'
+}
+const missScarlet = {
+  firstName: 'Scarlet',
+  lastName: 'Johansson',
+  color: 'red',
+  description: 'She is beautiful',
+  age: 30,
+  image: 'assets/scarlet.png',
+  occupation: 'Beauty'
+}
+const mrsWhite = {
+  firstName: 'Edna',
+  lastName: 'White',
+  color: 'white',
+  description: 'She knows everything',
+  age: 60,
+  image: 'assets/white.png',
+  occupation: 'Know it all'
+}
+
+// CREATE OBJECTS FOR ALL THE WEAPONS, ADD MORE CHARACTERISTICS TO THE WEAPONS IF YOU LIKE.
+
+const rope = {
+  name: 'rope',
+  weight: 5,
+  image: 'assets/rope.png'
+}
+const gun = {
+  name: 'gun',
+  weight: 40,
+  image: 'assets/gun.png'
+}
+const knife = {
+  name: 'knife',
+  weight: 20,
+  image: 'assets/knife.png'
+}
+const wrench = {
+  name: 'wrench',
+  weight: 50,
+  image: 'assets/wrench.png'
+}
+const candlestick = {
+  name: 'candlestick',
+  weight: 30,
+  image: 'assets/candle.png'
+}
+const pipe = {
+  name: 'pipe',
+  weight: 10,
+  image: 'assets/pipe.png'
+}
+
+
+// THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
+
+
+// NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
+
+const suspects = [
+  mrGreen,
+  mrMustard,
+  mrsPeacock,
+  professorPlum,
+  missScarlet,
+  mrsWhite
+]
+
+const weapon = [
+  rope,
+  gun,
+  knife,
+  wrench,
+  candlestick,
+  pipe
+]
+
+const room = [
+  'hall',
+  'lounge',
+  'study',
+  'library',
+  'kitchen',
+  'ballroom'
+]
+
+// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
+// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
+const randomSelector = array => {
+  return array[Math.floor(Math.random() * array.length)]
+}
+
+// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
+// With a killer, a weapon and a room.
+// The values will be set later.
+let crime = {
+ killer: [],
```

Comment:
Good assumption Claire. Yes, we shouldn't initialise those properties with empty array, because they are not going to be non-empty array later on. It's misrepresentation that could led to wrong assumption later on.

It's far more semantic approach to initialise mystery object with `null` values like this :

```
const mystery = {
  killer: null,
  room: null,
  weapon: null
}
```

Moreover, you can use `const` keyword - we will be changing values of properties, that's true, but JavaScript will not consider this as overwriting. What counts for JavaScript is that object will stay object